SU2_CFD/src/variables/CTransLMVariable.cpp

```diff
@@ -28,18 +28,37 @@
 
 #include "../../include/variables/CTransLMVariable.hpp"
 
-
-CTransLMVariable::CTransLMVariable(su2double intermittency, su2double REth, unsigned long npoint, unsigned long ndim,
-                                   unsigned long nvar, CConfig *config) : CTurbVariable(npoint, ndim, nvar, config) {
+CTransLMVariable::CTransLMVariable(su2double Intermittency, su2double ReThetaT, su2double gammaSep, su2double gammaEff, unsigned long npoint, unsigned long ndim, unsigned long nvar, CConfig *config) 
+  : CTurbVariable(npoint, ndim, nvar, config) {
 
   for(unsigned long iPoint=0; iPoint<nPoint; ++iPoint)
   {
-    Solution_Old(iPoint,0) = Solution(iPoint,0) = intermittency;
-    Solution_Old(iPoint,1) = Solution(iPoint,1) = REth;
+    Solution(iPoint,0) = Intermittency;
+    Solution(iPoint,1) = ReThetaT;
   }
 
-  if (config->GetMultizone_Problem())
-    Set_BGSSolution_k();
+  Solution_Old = Solution;
+
+  /*--- Setting CTransLMVariable of intermittency_Eff---*/
+  Intermittency_Eff.resize(nPoint) = gammaEff;
+  Intermittency_Sep.resize(nPoint) = gammaSep;
+  
+}
+
+void CTransLMVariable::SetIntermittencyEff(unsigned long iPoint, su2double val_Intermittency_sep) {
+
+  AD::StartPreacc();
+  AD::SetPreaccIn(Solution[iPoint], nVar);
+  /*--- Effective intermittency ---*/
+  Intermittency_Eff(iPoint) = max(Solution(iPoint,0),val_Intermittency_sep);
+  AD::SetPreaccOut(Intermittency_Eff(iPoint));
+  AD::EndPreacc();
```

Comment:
Pre accumulation is not needed for very simple computations like these.
```suggestion
  /*--- Effective intermittency ---*/
  Intermittency_Eff(iPoint) = max(Solution(iPoint,0), val_Intermittency_sep);
```